web/src/app/modules/views/main/editors/modules/contents-container/base/contents-container-base.ts

```diff
@@ -1,117 +1,119 @@
-        copy.description = model.description;
-        copy.id = Id.uuid;
-        copy.name = this.translate.instant('copyOf') + ' ' + model.name;
-        this.clipboardService.clipboard = [copy];
-    }
-
-    public async paste(name: string): Promise<void> {
-        if (!this.canPaste) {
-            return;
-        }
-        const model = this.clipboardModel;
-        const pasteName = name || this.clipboardModelName;
-        const copy = await this.createElement(pasteName);
-        copy.description = model.description;
-
-        const contents = await this.dataService.readContents(model.url, true);
-        const transformer = new GraphTransformer(this.dataService, undefined, copy);
-
-        const compoundId = Id.uuid;
-        await this.dataService.updateElement(copy, true, compoundId);
-        await transformer.cloneSubgraph(contents, compoundId, true, 0);
-        await this.dataService.commit(this.translate.instant('paste'));
-        await this.readContents();
-        this.clipboardService.clear();
-    }
-
-    public get canPaste(): boolean {
-        return this.clipboardService.clipboard.length === 1 && this.condition(this.clipboardService.clipboard[0]);
-    }
-}
+import { SpecmateDataService } from '../../../../../../data/modules/data-service/services/specmate-data.service';
+import { NavigatorService } from '../../../../../../navigation/modules/navigator/services/navigator.service';
+import { TranslateService } from '@ngx-translate/core';
+import { IContainer } from '../../../../../../../model/IContainer';
+import { ConfirmationModal } from '../../../../../../notification/modules/modals/services/confirmation-modal.service';
+import { Id } from '../../../../../../../util/id';
+import { OnInit, Input, Type } from '@angular/core';
+import { Objects } from '../../../../../../../util/objects';
+import { ClipboardService } from '../../tool-pallette/services/clipboard-service';
+import { GraphTransformer } from '../../tool-pallette/util/graph-transformer';
+
+export abstract class ContentContainerBase<T extends IContainer> implements OnInit {
+
+    protected abstract get condition(): (element: IContainer) => boolean;
+    protected contents: IContainer[];
+
+    private _parent: IContainer;
+
+    protected get parent(): IContainer {
+        return this._parent;
+    }
+
+    @Input()
+    protected set parent(parent: IContainer) {
+        this._parent = parent;
+        this.readContents();
+    }
+
+    constructor(
+        protected dataService: SpecmateDataService,
+        protected navigator: NavigatorService,
+        protected translate: TranslateService,
+        protected modal: ConfirmationModal,
+        private clipboardService: ClipboardService) {
+    }
+
+    public async ngOnInit(): Promise<void> {
+        await this.readContents();
+    }
+
+    public async create(name: string): Promise<void> {
+        let invalidCharacters: RegExp = new RegExp('[,;|]');

```

Comment:
Can you reuse the pattern configured in the field meta information, e.g. by FieldMeta.INamed.allowedPattern?

Reply:
```suggestion
        let validPattern = MetaInfo.INamed[0].allowedPattern;
```
is this alright?